text/safrole.tex

```diff
@@ -185,7 +185,7 @@ \subsection{The Slot Key Sequence}
   \end{aligned}\right.
 \end{equation}
 
-Finally, $F$ is the fallback key sequence function which selects an epoch's worth of validator Bandersnatch keys ($\seq{\mathbb{H}_B}_{\mathsf{E}}$) at random from the validator key set $\mathbf{k}$ using the entropy collected on-chain $r$:
+Finally, $F$ is the fallback key sequence function which selects an epoch's worth of validator Bandersnatch keys ($\seq{\mathbb{H}_B}_{\mathsf{E}}$) from the validator key set $\mathbf{k}$ using the entropy collected on-chain $r$:
```

Comment:
Removed "at random" which may give the impression to the reader that is really random (even though it should be obvious as we're talking about an onchain process). Maybe we can explicitly say pseudo-random :-)